### PR TITLE
Code Review Groups: implement unassign all button

### PR DESCRIPTION
**Pull request by bencodeorg**
On click of "Unassign All" button in Code Review Groups modal, move all students currently in groups into the unassigned area.

image

A couple of cleanup items as well:
- Cleans up header styling to match Figma.
- Deep clones groups before modifying them, as I realized that changes we were making to group membership was being made directly to the state object b/c the spread operator only shallow clones objects.
- Blank string for group name in newly created groups to make React happy.

## Testing story

Tested manually, and added a unit test.

### CODE REVIEW

File: apps/src/templates/codeReviewGroups/CodeReviewGroupsManager.jsx [hunk 1]
```diff
@@ -161,12 +185,12 @@ const addDroppableIdToGroup = group => {
 // Exported for a test
 export const getAssignedGroupDroppableId = id => DROPPABLE_ID_PREFIX + id;
 
-// TO DO: present a modal that allows a user to select a group name before creating it.
 // We need to generate a unique identifier for each group that is generated on the client
 // before we save it -- use a timestamp for this unique identifier.
 const generateNewGroup = () => {
   return {
     droppableId: `${DROPPABLE_ID_PREFIX}${new Date().getTime()}`,
+    name: '',
```
Comment (bencodeorg) [inline apps/src/templates/codeReviewGroups/CodeReviewGroupsManager.jsx:193]: Unrelated cleanup

File: apps/src/templates/codeReviewGroups/CodeReviewGroupsManager.jsx [hunk 2]
```diff
@@ -33,16 +33,37 @@ export default function CodeReviewGroupsManager({initialGroups}) {
...
+    });
+
+    const updatedAssignedGroups = getAssignedGroups().map(group => ({
+      ...group,
+      members: []
+    }));
+    setGroups([...updatedAssignedGroups, updatedUnassignedGroup]);
+  };
+
+  const unassignAllFromGroup = (
```
Comment (sanchitmalhotra126) [inline apps/src/templates/codeReviewGroups/CodeReviewGroupsManager.jsx]: nit: could this be something like `copyToUnassigned` maybe, just to be clear that this is only moving students to the unassigned group, not deleting them from another group?

Or alternatively, what if this function called `updateGroups` with the updated unassigned group and updated empty group and just returned the full list of groups? That way you'd be unassigning all from a single group in a single step?

**Review by sanchitmalhotra126 — COMMENTED** [review]
One nit/suggestion around the unassign logic, otherwise looks good!

File: apps/src/templates/codeReviewGroups/CodeReviewGroupsManager.jsx (hunk 2 above)
Comment (bencodeorg) [reply apps/src/templates/codeReviewGroups/CodeReviewGroupsManager.jsx]: Yeah I felt like the naming wasn't great here / logic a bit confusing, I think these are good ideas. Let me give it a shot.

Comment (bencodeorg) [reply apps/src/templates/codeReviewGroups/CodeReviewGroupsManager.jsx]: I renamed per your first suggestion -- the second idea makes sense to me, but I've been a little inconsistent in where I copy groups before changing them and would need to make a few changes to make that work. I am planning on pulling a lot of these get/change/etc methods out into a little util file, so I'm thinking I could maybe clean this up at that point? Makes sense to me though, as I'm looping through assigned groups twice

**Review by sanchitmalhotra126 — APPROVED** [review]
LGTM!

**Review by molly-moen — APPROVED** [review]
LGTM!
